Extension/src/githubAPI.ts

```diff
@@ -60,8 +60,11 @@ function isAsset(input: any): input is Asset {
  * @param input Incoming object.
  * @return Whether input is of type Build.
  */
-function isBuild(input: any): input is Build {
-    return input && input.name && typeof(input.name) === "string" && isArrayOfAssets(input.assets) && input.assets.length >= 3;
+function isBuild(input: any, isValid: boolean = true): input is Build {
+    // if isValid = true, check if the input is a valid build type with 3 assets,
+    // otherwise, check if the input is a build type.
+    return input && input.name && typeof (input.name) === "string" && isArrayOfAssets(input.assets) &&
+        (!isValid || input.assets.length >= 3);
 }
```

Comment:
I changed the functionality of the isBuild function, to recognize if an input is a "build" type, or if it is a "build" type with 3 assets (i.e. "valid" build type)

Reply:
I don't think we want to change this.  What we want is for `getReleaseJson` to return a filtered array of 5 or so `Build`s that are all valid.  So we just need to add a new function that gets 5 builds out of the json.  I don't think anything else has to change.

The other problem with doing it this way is that `getTargetBuildInfo` is going to have to do more checks to work correctly.  It currently exits early if the first build in the array matches the installed build, but doesn't check that the build is valid.  I think there is an expectation by all the logic involved that `getReleaseJson` should return a valid array of `Build`s.

Reply:
We talked about this offline. We do still want to include the invalid builds in the array, but we can stop adding entries once we see a "release" version.

Reply:
we are including the invalid builds. 
The behaviour of isBuild is as below:
IsValid === false, when the input has a name and the name is a type string, it is a **build**
IsValid === True, when the input has a name and it has 3 assets, it is a **Valid build**

Reply:
So first, I extract all builds (isValid=false) until I reach a valid release build,
and later on, I choose only the valid builds (isValid=true).